core/src/test/java/org/apache/iceberg/TableTestBase.java

```diff
@@ -349,6 +349,35 @@ void validateSnapshot(Snapshot old, Snapshot snap, long sequenceNumber, DataFile
     validateSnapshot(old, snap, (Long) sequenceNumber, newFiles);
   }
 
+  Snapshot commit(Table tbl, SnapshotUpdate snapshotUpdate, String branch) {
+    Snapshot snapshot = null;
+
+    if (branch.equals(SnapshotRef.MAIN_BRANCH)) {
+      snapshotUpdate.commit();
+      snapshot = tbl.currentSnapshot();
+    } else {
+      ((SnapshotProducer) snapshotUpdate.toBranch(branch)).commit();
```

Comment:
Why not just pass a `SnapshotProducer` to this method? That would eliminate casting right?

Reply:
MergeAppend doesn't implement SnapshotProducer